mesonbuild/mlog.py

```diff
@@ -45,19 +45,22 @@ def is_windows() -> bool:
     return platname == 'windows'
 
 def _windows_ansi() -> bool:
-    # windll only exists on windows, so mypy will get mad
-    from ctypes import windll, byref  # type: ignore
-    from ctypes.wintypes import DWORD
-
-    kernel = windll.kernel32
-    stdout = kernel.GetStdHandle(-11)
-    mode = DWORD()
-    if not kernel.GetConsoleMode(stdout, byref(mode)):
-        return False
-    # ENABLE_VIRTUAL_TERMINAL_PROCESSING == 0x4
-    # If the call to enable VT processing fails (returns 0), we fallback to
-    # original behavior
-    return bool(kernel.SetConsoleMode(stdout, mode.value | 0x4) or os.environ.get('ANSICON'))
+    if sys.platform == 'win32':
+        # windll only exists on windows, so mypy will get mad
+        from ctypes import windll, byref
+        from ctypes.wintypes import DWORD
+
+        kernel = windll.kernel32
+        stdout = kernel.GetStdHandle(-11)
+        mode = DWORD()
+        if not kernel.GetConsoleMode(stdout, byref(mode)):
+            return False
+        # ENABLE_VIRTUAL_TERMINAL_PROCESSING == 0x4
+        # If the call to enable VT processing fails (returns 0), we fallback to
+        # original behavior
+        return bool(kernel.SetConsoleMode(stdout, mode.value | 0x4) or os.environ.get('ANSICON'))
+    else:
+        assert False, 'This function should only be called on Windows platform'
```

Comment:
Unfortunately, mypy is not clever enough to disable the block following an assert...

Reply:
Yeah. It seems so, unfortunately: code written like this is really disgusting to look at.

Reply:
On windows, it complains about unused `type: ignore`...

Reply:
but if this is the only blocking part of this PR, I have no problems dropping the commit about mlog.

Reply:
Since `_windows_ansi()` is used in exactly one place, why not just refactor it into:
```python
def _should_colorize() -> bool:
     if not os.isatty(sys.stdout.fileno()):
        return False
    if is_windows():
        <implementation of _windows_ansi>
    else:
        return os.environ.get('TERM', 'dumb') != 'dumb'
```       
then in `colorize_console` you can just use that and not have to repeate the isatty check